quarkus/runtime/src/test/java/or/keycloak/quarkus/runtime/cli/PicocliTest.java

```diff
@@ -244,4 +240,85 @@ public void spiRuntimeWarnWithBuild() {
         assertThat(nonRunningPicocli.getOutString(), containsString("The following run time options were found, but will be ignored during build time: kc.spi-something-pass"));
     }
     
+    @Test
+    public void failIfOptimizedUsedForFirstStartupExport() {
+        NonRunningPicocli nonRunningPicocli = pseudoLaunch("export", "--optimized", "--dir=data");
+        assertEquals(CommandLine.ExitCode.USAGE, nonRunningPicocli.exitCode);
+        assertThat(nonRunningPicocli.getErrString(), containsString("The '--optimized' flag was used for first ever server start."));
+    }
```

Comment:
```suggestion
    @Test
    @NonRunningServer
    @Launch({"export", "--optimized", "--dir=data"})
    public void failIfOptimizedUsedForFirstStartupExport(CLIResult cliResult) {
        cliResult.assertExitCode(CommandLine.ExitCode.USAGE);
        cliResult.assertError("The '--optimized' flag was used for first ever server start.");
    }
```

I would like some sort of this ^^ approach. Not introducing any other `pseudoLaunch` and these kind of constructs on developer side, but rather have it hidden.

-  The `NonRunningPicocli` might be some kind of `KeycloakDistribution` type that would not start the server
- We could leverage the `@Launch` annotation
- We could leverage the `KeycloakDistribution` parameter injection
- The `@NonRunningServer` (or diff name) annotation would mark the test to not start server itself
- In that case, we would not have to rewrite tests so much.

@shawkins WDYT?

Reply:
My concerns with going this approach are:
- not as good at localizing the tests - they will be in the test module, not in the quarkus-server module
- additional / alternative setup is required to test scenarios dependent up augmented state
- devs will have to be very careful to not hit the subtle differences, such as:
  - the error/out contain console characters
  - the execution is not via the shell scripts, which contains a lot of manipulation
  - the KeycloakDistribution would be mostly not implemented
  
I'd probably be more comfortable with adding support for a option that would cause the distribution to act more like a non-running server. That is if -Dkc.test.non-running or something is specified then we'll inhibit reaug and skip full exection of the command.